apps/frontend/src/pages/CreateCharity.tsx

```diff
@@ -0,0 +1,651 @@
+import { ChangeEvent, FormEventHandler, useState } from "react";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Checkbox,
+  Container,
+  FormControl,
+  FormControlLabel,
+  Grid,
+  Paper,
+  TextField,
+  Typography,
+} from "@mui/material";
+import SearchBar from "../components/searchBar";
+import { colors } from "../styles/colors";
+import _axios_instance from "../_axios_instance";
+import { CharityObject, OrganizationObject } from "../types.ts";
+
+interface ImageObject {
+  image: File | undefined;
+  imageString: string | undefined;
+  index: number;
+}
+
+interface ImageUploadObject {
+  image: File;
+  index: number;
+}
+
+interface ImageUploadedObject {
+  url: string;
+  index: number;
+}
+
+const CreateCharity = () => {
+  const [sent, setSent] = useState(false);
+  const [startDate, setStartDate] = useState<string>("2024-07-15");
+  const [partnerImages, setPartnerImages] = useState<ImageObject[]>([]);
+  const [logoImage, setLogoImage] = useState<File | null>(null);
+  const [logoImageString, setLogoImageString] = useState<string>();
+
+  const [newCharityObject, setNewCharityObject] = useState<CharityObject>({
+    name: "",
+    description: "",
+    startDate: "",
+    endDate: "",
+    imageUrl: "",
+    organizations: [],
+  });
+
+  // Submits
+  const handleSubmit: FormEventHandler<HTMLFormElement> = async (
+    submissionEvent,
+  ) => {
+    submissionEvent.preventDefault();
+    if (sent) return;
+    const logoURL = await asyncListingImageWrapperLogo();
+    if (!logoURL) {
+      alert("Image failed to upload");
+      return;
+    }
+    const newDataObject = await asyncListingImageWrapperPartners(logoURL);
+    if (newDataObject) {
+      try {
+        const response = await _axios_instance.post(
+          "/charities",
+          newDataObject,
+        );
+        if (response) {
+          alert("Charity Created!");
+          setSent(true);
+        } else {
+          alert("Charity Creation Failed");
+          setSent(false);
+        }
+      } catch (error) {
+        alert("Charity Creation Failed");
+        console.error(error);
+        setSent(false);
+      }
+    } else {
+      alert("Images failed to upload, please try again later");
+      setSent(false);
+    }
+  };
```

Comment:
Rather than using a "sent" variable to track the status of the request, it might be better to clear all the form fields once a successful request has gone through. Having another variable just adds room for more bugs. I think a setSent(false) is missed if there is no logoURL

Reply:
Fixed